code/extensions/QuickFeedbackExtension.php

```diff
@@ -1,47 +1,99 @@
 <?php
 
 class QuickFeedbackExtension extends DataExtension {
-
+	/**
+	 * @var array
+	 */
 	private static $allowed_actions = array(
 		'QuickFeedbackForm'
 	);
 
 	/**
-	 *	Feedback Form
+	 * @return Form
 	 */
 	public function QuickFeedbackForm() {
 		$form = Form::create(
 			$this->owner,
 			'QuickFeedbackForm',
 			FieldList::create(
-				LiteralField::create("RatingTitle", "<h4>Was this article helpful?</h4>"),
+				LiteralField::create('RatingTitle', _t('QuickFeedback.Title', '<h4>Was this article helpful?</h4>')),
 				ButtonGroupField::create('Rating', '', array(
-					"1" => "Yes",
-					"0" => "No",
+					'1' => _t('QuickFeedback.Yes', 'Yes'),
+					'0' => _t('QuickFeedback.No', 'No'),
 				)),
-				TextareaField::create('Comment','Comment')
+				TextareaField::create('Comment', _t('QuickFeedback.Comment', 'Comment'))
 			),
 			FieldList::create(
-				FormAction::create('doSubmit','Submit')
+				FormAction::create('doSubmit', _t('QuickFeedback.Submit', 'Submit'))
 			)
 		);
 
 		return $form;
-	}	
+	}
 
 	/**
-	 *	Submits feedback form
+	 * @param array $data
+	 * @param Form $form
+	 *
+	 * @return mixed
 	 */
 	public function doSubmit($data, $form) {
+		$controller = Controller::curr();
+
+		if (!$controller) {
+			goto error;
+		}
+
+		$request = $controller->getRequest();
+
+		if (!$request) {
+			goto error;
+		}
+
+		$limit = Config::inst()->get('QuickFeedbackExtension', 'rate_limit');
+
+		if (!$limit) {
+			$limit = 3;
+		}
+
+		$existing = Feedback::get()
+			->filter('IP', $request->getIP())
+			->sort('Created desc')
+			->first();
+
+		if ($existing) {
+			$created = $existing->dbObject('Created');
+
+			if (!$created) {
+				goto error;
+			}
+
+			$words = $created->TimeDiffIn('minutes');
+			$parts = explode(' ', $words);
+
+			$minutes = (int) $parts[0];
+
+			if ($minutes <= $limit) {
+				goto rate;
```

Comment:
if goto compatible with php 5.3.3... perhaps we should specify the php version required in composer.json (considering the 3.x core requires 5.3.3 compatibility to function)